LogMessageWaitStrategy fails in hollow mode with NPE, assuming a container log

**Describe the bug**

In "hollow" mode, using a config like:

```java
   public class BonusPayoutITContainerConfig implements SharedContainerConfiguration {

    @Container
    public static ApplicationContainer inventory = new ApplicationContainer()
                    .withAppContextRoot("/")
                    .withExposedPorts(9443)
                    .waitingFor(
                            Wait.forLogMessage(".*CWWKF0011I.*\\n", 1)
                        );
   }
```

fails with:
```
java.util.concurrent.CompletionException: java.lang.NullPointerException: containerId was not specified
Caused by: java.lang.NullPointerException: containerId was not specified
```

with stack trace: 

> Preconditions.checkNotNull(T, Object) line: 228	
> LogContainerCmdImpl.withContainerId(String) line: 78	
> LogContainerCmdImpl.<init>(LogContainerCmd$Exec, String) line: 38	
> DockerClientImpl.logContainerCmd(String) line: 380	
> AuditLoggingDockerClient.logContainerCmd(String) line: 26	
> LogUtils.attachConsumer(DockerClient, String, Consumer<OutputFrame>, boolean, OutputType...) line: 88	
> LogUtils.followOutput(DockerClient, String, Consumer<OutputFrame>, OutputType...) line: 36	
> LogUtils.followOutput(DockerClient, String, Consumer<OutputFrame>) line: 51	
> LogMessageWaitStrategy.waitUntilReady() line: 22	
> LogMessageWaitStrategy(AbstractWaitStrategy).waitUntilReady(WaitStrategyTarget) line: 35	
> ApplicationContainer(GenericContainer<SELF>).waitUntilContainerStarted() line: 879	
> ApplicationContainer.doStart() line: 296	
> ApplicationContainer(GenericContainer<SELF>).start() line: 300	

Assuming we probably need to either:
* figure we don't need to wait, because it's hollow mode, and things are already started.
 OR
* use the "hollow" JVM's log stream to search for the log message.

**To Reproduce**
Using Open Liberty with the v0.9 adapater, using a config like above, run:   `mvn liberty:dev`

**Expected behavior**
Either no-op the wait, or execute wait based on the "hollow" JVM which doesn't need to be started.

**System information:**
 - OS: Windows
 - Java Version:  11
 - MicroShed Testing Version:  **org.microshed:microshed-testing-liberty:jar:0.9:**

--- a/.github/workflows/microshed-ci.yml
+++ b/.github/workflows/microshed-ci.yml
@@ -1,10 +1,6 @@
 name: MicroShed CI
 
 on:
-  push:
-    branches: 
-    - master
-    - gh-actions
   pull_request:
     branches: 
     - master

--- a/modules/testcontainers/src/main/java/org/microshed/testing/testcontainers/ApplicationContainer.java
+++ b/modules/testcontainers/src/main/java/org/microshed/testing/testcontainers/ApplicationContainer.java
@@ -54,9 +54,12 @@ import org.microshed.testing.testcontainers.internal.HollowContainerInspection;
 import org.microshed.testing.testcontainers.internal.ImageFromDockerfile;
 import org.microshed.testing.testcontainers.spi.ServerAdapter;
 import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.BindMode;
 import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.SelinuxContext;
 import org.testcontainers.containers.output.Slf4jLogConsumer;
 import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
 import org.testcontainers.containers.wait.strategy.Wait;
 import org.testcontainers.containers.wait.strategy.WaitStrategy;
 import org.testcontainers.utility.Base58;
@@ -435,13 +438,21 @@ public class ApplicationContainer extends GenericContainer<ApplicationContainer>
     @Override
     public ApplicationContainer waitingFor(WaitStrategy waitStrategy) {
         waitStrategySet = true;
+        checkHollowWaitStrategy(waitStrategy);
         return super.waitingFor(waitStrategy);
     }
 
     @Override
     public void setWaitStrategy(WaitStrategy waitStrategy) {
         waitStrategySet = true;
+        checkHollowWaitStrategy(waitStrategy);
         super.setWaitStrategy(waitStrategy);
+    }
+
+    private void checkHollowWaitStrategy(WaitStrategy strat) {
+        if (isHollow && waitStrategy instanceof LogMessageWaitStrategy) {
+            throw new UnsupportedOperationException("WaitStrategy [" + strat + "] is not supported in hollow container mode");
+        }
     }
 
     /**
@@ -535,6 +546,35 @@ public class ApplicationContainer extends GenericContainer<ApplicationContainer>
         }
         super.withReuse(reusable);
         return this;
+    }
+
+    @Override
+    public ApplicationContainer withFileSystemBind(String hostPath, String containerPath) {
+        requireNonHollow();
+        return super.withFileSystemBind(hostPath, containerPath);
+    }
+
+    @Override
+    public ApplicationContainer withFileSystemBind(String hostPath, String containerPath, BindMode mode) {
+        requireNonHollow();
+        return super.withFileSystemBind(hostPath, containerPath, mode);
+    }
+
+    @Override
+    public void addFileSystemBind(String hostPath, String containerPath, BindMode mode) {
+        requireNonHollow();
+        super.addFileSystemBind(hostPath, containerPath, mode);
+    }
+
+    @Override
+    public void addFileSystemBind(String hostPath, String containerPath, BindMode mode, SelinuxContext selinuxContext) {
+        requireNonHollow();
+        super.addFileSystemBind(hostPath, containerPath, mode, selinuxContext);
+    }
+
+    private void requireNonHollow() {
+        if (isHollow)
+            throw new UnsupportedOperationException("Operation not supported in hollow container mode");
     }
 
     /**

